### PR TITLE
[3190] Add course resource to v2

### PR DESCRIPTION
**Pull request by AgaDufrat**
### Context
Changes are required for the Accredited body feature in Publish.

### Changes proposed in this pull request

Allow retrieving courses across multiple training providers in one
hit. Specifically for accredited bodies to be able to see the courses of
their training providers.

Having this as a new resource-style endpoint with the ability to apply
filters will allow us to consolidate all the other course endpoints.

### Guidance to review
- Generate a token: TOKEN=`bin/mcb apiv2 token generate -S secret user@example.com`
- `curl -H 'Accept: application/json' -H "Authorization: Bearer ${TOKEN}" 'http://localhost:3001/api/v2/recruitment_cycles/2020/courses?filter%5Baccrediting_provider_code%5D%3DE69%27’`
### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by AgaDufrat** [comment]
[[S] Add courses endpoint that supports filters and spares fieldsets](https://trello.com/c/tTR5M5Au/3190-add-courses-endpoint-that-supports-filters-and-spares-fieldsets)

File: spec/requests/api/v2/courses_spec.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+  let(:current_cycle) { find_or_create :recruitment_cycle }
+  let(:next_cycle)    { find_or_create :recruitment_cycle, :next }
+  let(:current_year)  { current_cycle.year.to_i }
+  let(:previous_year) { current_year - 1 }
+  let(:next_year)     { current_year + 1 }
+  let(:subjects) { [course_subject_mathematics] }
+
+  let(:providerA)       { create :provider, organisations: [organisation] }
```
Comment (asmega) [inline spec/requests/api/v2/courses_spec.rb]: snake_case? not sure why the change to camelCase here?

File: spec/requests/api/v2/courses_spec.rb [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      response
+    end
+
+    context "request without filters" do
+      let(:request_path) do
+        api_v2_recruitment_cycle_courses_path(current_cycle.year) +
+          "?include=subjects"
+      end
+
+      it { should have_http_status(:success) }
```
Comment (asmega) [inline spec/requests/api/v2/courses_spec.rb]: personally i would merge this test into the test below. as these tests are expensive and there's no real gain to have it split out. it'll save the test setup and request

File: spec/requests/api/v2/courses_spec.rb (hunk 1 above)
Comment (AgaDufrat) [reply spec/requests/api/v2/courses_spec.rb]: I changed it to use numbers since we seem to do it in other tests.
